database/manager/manager.go

```diff
@@ -0,0 +1,252 @@
+		if err != nil {
+			return err
+		}
+
+		// If [version] is greater than or equal to the specified version
+		// skip over creating the new database to avoid creating the same db
+		// twice or creating a database with a version ahead of the desired one.
+		if cmp := version.Compare(currentVersion); cmp >= 0 {
+			return filepath.SkipDir
+		}
+
+		db, err := leveldb.New(path, 0, 0, 0)
+		if err != nil {
+			return fmt.Errorf("couldn't create db at %s: %w", path, err)
+		}
+
+		semDBs = append(semDBs, &VersionedDatabase{
+			Database: db,
+			Version:  version,
+		})
+
+		return filepath.SkipDir
+	})
+
+	SortDescending(semDBs)
+
+	m := &manager{
+		databases: semDBs,
+	}
+
+	// If an error occurred, close all of the opened databases
+	// and return the original error
+	if err != nil {
+		_ = m.Close()
+		return nil, err
+	}
+
+	return m, nil
+}
+
+// NewPrefixDBManager creates a new manager with each database instance prefixed
+// by [prefix]
+func (m *manager) NewPrefixDBManager(prefix []byte) Manager {
+	m, _ = m.wrapManager(func(sdb *VersionedDatabase) (*VersionedDatabase, error) {
+		return &VersionedDatabase{
+			Database: prefixdb.New(prefix, sdb.Database),
+			Version:  sdb.Version,
+		}, nil
+	})
+	return m
+}
+
+// NewNestedPrefixDBManager creates a new manager with each database instance
+// wrapped with a nested prfix of [prefix]
+func (m *manager) NewNestedPrefixDBManager(prefix []byte) Manager {
+	m, _ = m.wrapManager(func(sdb *VersionedDatabase) (*VersionedDatabase, error) {
+		return &VersionedDatabase{
+			Database: prefixdb.NewNested(prefix, sdb.Database),
+			Version:  sdb.Version,
+		}, nil
+	})
+	return m
+}
+
+// NewMeterDBManager wraps each database instance with a meterdb instance. The namespace
+// is concatenated with the version of the database. Note: calling this more than once
+// with the same [namespace] will cause a conflict error for the [registerer]
+func (m *manager) NewMeterDBManager(namespace string, registerer prometheus.Registerer) (Manager, error) {
+	return m.wrapManager(func(sdb *VersionedDatabase) (*VersionedDatabase, error) {
+		mdb, err := meterdb.New(fmt.Sprintf("%s_%s", namespace, strings.ReplaceAll(sdb.Version.String(), ".", "_")), registerer, sdb.Database)
+		if err != nil {
+			return nil, err
+		}
+		return &VersionedDatabase{
+			Database: mdb,
+			Version:  sdb.Version,
+		}, nil
+	})
+}
+
+// NewManagerFromDBs
+func NewManagerFromDBs(dbs []*VersionedDatabase) (Manager, error) {
+	SortDescending(dbs)
+	sortedAndUnique := utils.IsSortedAndUnique(innerSortDescendingVersionedDBs(dbs))
+	if !sortedAndUnique {
+		return nil, errNonSortedAndUniqueDBs
+	}
+	return &manager{
+		databases: dbs,
+	}, nil
+}
+
+type VersionedDatabase struct {
+	database.Database
+
+	version.Version
+}
+
+type innerSortDescendingVersionedDBs []*VersionedDatabase
+
+// Less returns true if the version at index i is greater than the version at index j
+// such that it will sort in descending order
```

Comment:
I would add `// such that it will sort in descending order (newest version --> oldest version)`